### DISCUSSION TITLE
Unable to compile when importing, otherwise it works fine.

### DISCUSSION THREAD

**Discussion by hadiimran00** [crytic/slither#1745] [Q&A]
### Describe the issue:

Program does not run/compile when I import otherwise it works fine in VS code.
![image](https://user-images.githubusercontent.com/111147135/223413629-17a7b98b-3d2b-419e-91cd-e70f80624691.png)

### Code example to reproduce the issue:

// SPDX-License-Identifier: MIT

pragma solidity ^0.8.0;

import '@openzeppelin/contracts/token/ERC721/ERC721.sol';
import "@openzeppelin/contracts/utils/Counters.sol";

contract MyNFT is ERC721 {
    using Counters for Counters.Counter;
    Counters.Counter private _tokenIds;

    constructor() ERC721("MyNFT", "NFT") {}

    function mintNFT(address recipient, string memory tokenURI) public returns (uint256) {
        _tokenIds.increment();
        uint256 newItemId = _tokenIds.current();
        _mint(recipient, newItemId);
        _setTokenURI(newItemId, tokenURI);
        return newItemId;
    }
}

### Version:

0.9.2

**Comment by elopez** [comment]
Hi! Thanks for the report. What compilation framework (hardhat, foundry, etc) is your project using?

**Comment by hadiimran00** [comment]
> Hi! Thanks for the report. What compilation framework (hardhat, foundry, etc) is your project using?

hardhat

**Comment by elopez** [comment]
It does not seem to be noticing hardhat is present, do you have a `hardhat.config.js` or `hardhat.config.ts` file in that folder (`C:\Users\pc`)? Is that your project folder? (maybe you forgot to `cd` to your project folder before running slither?) Does it work if you force the use of hardhat? `slither . --compile-force-framework hardhat`

**Comment by hadiimran00** [comment]
> It does not seem to be noticing hardhat is present, do you have a `hardhat.config.js` or `hardhat.config.ts` file in that folder (`C:\Users\pc`)? Is that your project folder? (maybe you forgot to `cd` to your project folder before running slither?) Does it work if you force the use of hardhat? `slither . --compile-force-framework hardhat`

where can i get this file?

**Comment by elopez** [comment]
That file is part of your project if you're using Hardhat.

**Comment by hadiimran00** [comment]
> That file is part of your project if you're using Hardhat.

there is no such file in my directory

**Comment by elopez** [comment]
Are you sure you're in the correct directory then? That file is contained in all hardhat projects 🤔  `C:\Users\pc` doesn't sound like a project folder.

**Comment by hadiimran00** [comment]
> Are you sure you're in the correct directory then? That file is contained in all hardhat projects 🤔 `C:\Users\pc` doesn't sound like a project folder.

i didnt have made any specific folder all the files are in `C:\Users\pc`

**Comment by elopez** [comment]
so if you run `npx hardhat compile` there in `C:\Users\pc` it works for you and builds your project? or how do you build the project without a hardhat.config.js file? 🤔

**Reply by hadiimran00** [reply]
>
![image](https://user-images.githubusercontent.com/111147135/224448156-ab5fad40-609b-4ffe-8ea4-36b3734f1e37.png)

**Reply by hadiimran00** [reply]
![image](https://user-images.githubusercontent.com/111147135/224448300-560a11a0-daa4-4452-8f82-85862fbaaf02.png)

**Reply by hadiimran00** [reply]
I have to run slither on my smart contract, nothing else

**Reply by elopez** [reply]
You need to have a working project that is buildable first, before running Slither. Once you have something that hardhat can compile you'll be able to run Slither on it.

If you want to build just the code in a `.sol` file without using hardhat or any other framework, you can do so as well, but it will require significantly more effort -- you'll need to download any dependencies (like openzeppelin contracts) yourself, figure out and configure appropriate solc remappings, install an appropriate solc compiler, and then run slither on the code file.

**Comment by hadiimran00** [comment]
![image](https://user-images.githubusercontent.com/111147135/224510999-f9dcd02e-0235-424b-bd02-31b83f8314ce.png)
why it is saying nothing to compile

**Reply by elopez** [reply]
Do you have any solidity files in the hardhat contracts folder? That message is from hardhat, not slither.

**Comment by hadiimran00** [comment]
![image](https://user-images.githubusercontent.com/111147135/224513148-d491d08f-f142-4eac-9be5-d91997e56207.png)
I have made the changes in the code it keeps giving same old error

**Reply by elopez** [reply]
That's an error on your solidity code in `contracts/aa.sol`, `contract Myis ERC20` is not valid solidity syntax. Maybe you meant to write `contract My is ERC20` or something like that. Take a look at the solidity syntax for inheritance: https://docs.soliditylang.org/en/v0.8.17/contracts.html#inheritance

**Reply by hadiimran00** [reply]
I have made the changes in the code it keeps giving same old error

**Reply by hadiimran00** [reply]
Thank You! I was editing the wrong contract, It is working now.
